service/app/org/sunbird/telemetry/dispatcher/EkstepTelemetryDispatcher.java

```diff
@@ -0,0 +1,89 @@
+package org.sunbird.telemetry.dispatcher;
+
+import com.google.common.net.HttpHeaders;
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.request.BaseRequest;
+import com.typesafe.config.Config;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.RestUtil;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.util.ConfigUtil;
+import util.Constant;
+import util.Message;
+
+/**
+ * Dispatcher responsible for storing Telemetry in Ekstep platform.
+ *
+ * @author Manzarul
+ */
+public class EkstepTelemetryDispatcher {
+  private static Config config = ConfigUtil.getConfig();
+
+  public static boolean dispatch(Map<String, String[]> reqHeaders, String body) throws Exception {
+    Map<String, String> headers = getHeaders(reqHeaders);
+    BaseRequest request = Unirest.post(telemetryAPIURL()).headers(headers).body(body);
+    executeRequest(request);
+    return true;
+  }
+
+  public static boolean dispatch(Map<String, String[]> reqHeaders, byte[] body) throws Exception {
+    Map<String, String> headers = getHeaders(reqHeaders);
+    BaseRequest request = Unirest.post(telemetryAPIURL()).headers(headers).body(body);
+    executeRequest(request);
+    return true;
+  }
+
+  private static Map<String, String> getHeaders(Map<String, String[]> headers) {
```

Comment:
Why do we need this method? Are the headers that should be sent to EkStep fixed? We can avoid this computation on every request. All we need is two fixed sets of headers - one for JSON body and one for gzip requests.

Reply:
I removed Computation logic and based on request Content Type (json or gzip) , that method is returning values.